app/models/spree_adyen/gateway.rb

```
@@ -80,19 +80,22 @@ def cancel(id, payment)
       payload = SpreeAdyen::RefundPayloadPresenter.new(
         payment: payment,
         amount_in_cents: amount_in_cents,
         payment_method: self,
-        currency: payment.currency
+        currency: payment.currency,
+        refund: refund
       ).to_h
 
       response = send_request do
-        client.checkout.modifications_api.refund_captured_payment(payload, payment_id, headers: { 'Idempotency-Key' => SecureRandom.uuid })
+        client.checkout.modifications_api.refund_captured_payment(payload, payment.transaction_id, headers: { 'Idempotency-Key' => SecureRandom.uuid })
```

RuboCop (1.81.1)

[convention] Line is too long. [151/150]

(Layout/LineLength)